shell/platform/linux/fl_window_state_monitor_test.cc

```diff
@@ -0,0 +1,255 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/linux/fl_window_state_monitor.h"
+#include "flutter/shell/platform/linux/public/flutter_linux/fl_string_codec.h"
+#include "flutter/shell/platform/linux/testing/fl_test.h"
+#include "flutter/shell/platform/linux/testing/mock_binary_messenger.h"
+#include "flutter/shell/platform/linux/testing/mock_window.h"
+
+#include "gtest/gtest.h"
+
+class LifecycleStringMatcher {
+ public:
+  using is_gtest_matcher = void;
+
+  explicit LifecycleStringMatcher(::testing::Matcher<std::string> value)
+      : value_(std::move(value)) {}
+
+  bool MatchAndExplain(GBytes* data,
+                       ::testing::MatchResultListener* result_listener) const {
+    g_autoptr(FlStringCodec) codec = fl_string_codec_new();
+    g_autoptr(GError) error = nullptr;
+    g_autoptr(FlValue) value =
+        fl_message_codec_decode_message(FL_MESSAGE_CODEC(codec), data, &error);
+    if (value == nullptr) {
+      *result_listener << ::testing::PrintToString(error->message);
+      return false;
+    }
+    if (!value_.MatchAndExplain(fl_value_get_string(value), result_listener)) {
+      *result_listener << " where the value doesn't match: \"" << value << "\"";
+      return false;
+    }
+    return true;
+  }
+
+  void DescribeTo(std::ostream* os) const {
+    *os << "value ";
+    value_.DescribeTo(os);
+  }
+
+  void DescribeNegationTo(std::ostream* os) const {
+    *os << "value ";
+    value_.DescribeNegationTo(os);
+  }
+
+ private:
+  ::testing::Matcher<std::string> value_;
```

Comment:
Should this class be tagged with `FML_DISALLOW_COPY_AND_ASSIGN`?

Reply:
Again, cargo-culted and no other code in the Linux embedder has this tag, so maybe? I'll omit for now but would be worth checking the other cases if this should be added.